Diseño template Principals

MaksiConGit opened the pull request:
Diseño de template funcional. Falta show.

MaksiConGit committed:
feat<back-front>: se añaden bocetos de diseño de asistencia y materias

MaksiConGit committed:
feat<back-front>: se añaden bocetos de diseño de asistencia y materias

MaksiConGit committed:
feat<back-front>: se añaden bocetos de diseño de asistencia y materias

MaksiConGit committed:
feat<back-front>: se añaden bocetos de diseño de asistencia y materias

Marzopax committed:
feat<front>: horarios + tablas

tablas editables

Marzopax committed:
feat<front>: horarios + tablas

tablas editables

Marzopax committed:
feat<front>: horarios + tablas

tablas editables

Marzopax committed:
feat<front>: horarios + tablas

tablas editables

EstefaniSaiquita committed:
feat<front>: se crea componentes, views de profesores y alumnos

EstefaniSaiquita committed:
feat<front>: se crea componentes, views de profesores y alumnos

EstefaniSaiquita committed:
feat<front>: se crea componentes, views de profesores y alumnos

EstefaniSaiquita committed:
<front> view de material hecho

EstefaniSaiquita committed:
<front> view de material hecho

EstefaniSaiquita committed:
<front> view de material hecho

MaksiConGit committed:
Merge branch '154-diseño-template-students' of https://github.com/MaksiConGit/proyecto-final-tercero into 154-diseño-template-students

MaksiConGit committed:
feat<back-front>: index, edit, create funcional con diseño

MaksiConGit committed:
feat<back-front>: index, edit, create funcional con diseño

MaksiConGit committed:
feat<back-front>: directivo index, edit, creat funcional diseño

MaksiConGit committed:
feat<back-front>: directivo index, edit, creat funcional diseño

LucasDLabo committed:
Instalado Livewire

LucasDLabo committed:
feat <back-front>: Directivo create, index, edit

 - Ahora los principals se encuentras paginados y ordenados según id.
 - El create/edit de principals permite seleccionar una institución para crear su usuario.
 - Ahora el create de principals genera automáticamente un usuario con su nombre, apellido y email.
 - Se creo un Dev Table 'logindata' que almacena los datos de inicio de sesión sin la contraseña hasheada.
 - Ahora se permite seleccionar un país, provincia y ciudad.
 - Se creo un componente de blade nuevo 'input-email'.
 - Se cambio la migration de users de la columnda 'role_id' para que tenga un valor default y no lance error.

LucasDLabo committed:
feat <back-front>: Directivo create, index, edit

 - Ahora los principals se encuentras paginados y ordenados según id.
 - El create/edit de principals permite seleccionar una institución para crear su usuario.
 - Ahora el create de principals genera automáticamente un usuario con su nombre, apellido y email.
 - Se creo un Dev Table 'logindata' que almacena los datos de inicio de sesión sin la contraseña hasheada.
 - Ahora se permite seleccionar un país, provincia y ciudad.
 - Se creo un componente de blade nuevo 'input-email'.
 - Se cambio la migration de users de la columnda 'role_id' para que tenga un valor default y no lance error.

LucasDLabo committed:
Implementado Livewire, Roles y Permisos, Login

 - Se instaló Livewire.
 -  Se instaló Spatie Roles y Permisos.
 - Se crearon migraciones polimórficas.
 - Se implemento el login dentro del template.
 - Se crearon usuarios con datos reales para directivo, profesor y estudiante.

LucasDLabo committed:
feat: Botón de login/register y Perfil

 - Se implementó un condicional para logearse/registrarse o mostrar el perfil si ya se esta logeado.

LucasDLabo committed:
fix: user_id de Student Seeder

LucasDLabo reviewed (changes requested):
**Buen diseño, me gusta**
Posibles cambios a realizar (Algunas son ideas por lo que pueden no hacerse o ser de baja prioridad):

### Principal Index
- [ ] Si es una lista general, mostrar todas los profesores paginados y permitir filtrar por carrera, materia y curso.
- [ ] Idea: Si la card expresa "Usuario: Sin asignar" , mostrar un botón para cada uno que lleve a la asignación.
- [ ]  Idea: Para mejorar la experiencia de usuario, agregar un efecto de hover al pasar por encima de un profesor.
- [ ]  No esta listado en la navbar de la derecha la sección de "Directivos".
- [ ]  Idea: Implementar en la lista de directivos, opciones para editar y eliminar sin tener que acceder a cada directivo.
- [ ]  Idea: El botón de crear directivo puede pasar un poco desapercibido.
### Principal Create/Edit
- [ ]  El create/edit debe permitir como opción requerida elegir Institución para crear/editar el directivo y su relación correspondiente con la institutions_principal.
- [ ]  El botón de create/edit dice 'enviar' en vez de crear/editar.
- [ ]  El usuario deberá crearse automáticamente.
- [ ]  Idea: Al momento de crear el directivo, redirigir al show del mismo.
- [ ]  Los iconos de los inputs debe estar acorde a lo que pide el input.
- [ ] El Principal Edit tiene como título "Crear Profesor".
- [ ] La confirmación de Crear Directivo y Editar Directivo tiene errores de escritura.
- [ ] La confirmación de Crear Directivo y Editar Directivo tiene errores de escritura.

### Principal Show
- [ ]  El botón de eliminar deberá tener una confirmación de eliminación.
- [ ] Implementar Diseño para principal.show.
- [ ] Mostrar las instituciones a la que pertenece el Directivo
- [ ] Idea: Implementar un botón para volver hacia atrás.
### General
- [ ]  El breadcrum deberá funcionar correctamente en todas las páginas.
- [ ] El titulo de la pestaña debe estar acorde a la vista.

LucasDLabo committed:
feat: Create Principal Tabla Polimorfica

 - Ahora el user de principal se crea correctamente con los datos de las columnas polimórficas.
 - Se hizo un rebase de code.

LucasDLabo committed:
Merge branch '160-diseño-template-principals' of https://github.com/MaksiConGit/proyecto-final-tercero into 160-diseño-template-principals

LucasDLabo committed:
feat: Principal Index search y filtro

 - Se implemento búsqueda de directivos por barra buscadora y filtrado por instituciones.
